### PR TITLE
add test for pipeline run using .krmignore file

### LINKED ISSUE #1414: Rationalize krmignore with the filtering resources in `pipeline run`
We are filtering out resources during pipeline execution for ex. Kptfile, function configs etc. We need to investigate and support `krmignore` use-cases as well. Creating this for us to investigate first what needs to be done and then implement it.

### PR DESCRIPTION
**Pull request by natasha41575**
adds a test for pipeline run using krmignore

same test as https://github.com/GoogleContainerTools/kpt/pull/1453 but ignores the resource file, which means there should be no diff 

I believe the functionality is already implemented (see comment on https://github.com/GoogleContainerTools/kpt/issues/1414)

close #1414

### CODE REVIEW

**Review by Shell32-Natsu — COMMENTED** [review]
nit: No need to put all the files in `config` directory.

**Comment by natasha41575** [comment]
@Shell32-Natsu moving the files results in unexpected behavior, looking into it now

**Comment by Shell32-Natsu** [comment]
@natasha41575 I guess you need to add `.expect` to the `krmignore` as well.

**Comment by natasha41575** [comment]
@Shell32-Natsu done

File: internal/pipeline/testdata/packagewithkrmignore/Kptfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: app
+pipeline:
+  mutators:
+  - image: gcr.io/kpt-fn/set-namespace
+    configMap:
+      namespace: staging
+  - image: gcr.io/kpt-fn/set-label
+    configMap:
+      tier: backend
```
Comment (Shell32-Natsu) [inline internal/pipeline/testdata/packagewithkrmignore/Kptfile:12]: Just a reminder, currently set-label function doesn't support this input from `configMap` field. Will fix it soon. https://github.com/GoogleContainerTools/kpt/issues/1487

**Review by Shell32-Natsu — APPROVED** [review]
LGTM

File: internal/pipeline/testdata/packagewithkrmignore/Kptfile (hunk 1 above)
Comment (natasha41575) [reply internal/pipeline/testdata/packagewithkrmignore/Kptfile:12]: thanks!

**Pull request merged**